### ISSUE TITLE
Question: Is google spreedsheet api v4 supported?

### ISSUE THREAD

**Issue by jimi75** [simile-widgets/exhibit#196]
We have a project that parses a google spreadsheet with API v3. From Agust 2 2021 Google turn off v3. 

Exhibit support v4? 

Reference: https://cloud.google.com/blog/products/g-suite/migrate-your-apps-use-latest-sheets-api

**Comment by lavinesp** [comment]
I also have a project that parses a Google SS with API v3.  This project involves years of research and development.  Will Exhibit 3.0 support v4? If so, where can I find some documentation?

**Comment by karger** [comment]
We don't have people updating the exhibit codebase right now.  As it's 
an open source project, we are happy to accept patches from other people.

Our current project, Mavo ( http://mavo.io/ ) is significantly more 
powerful that exhibit in many ways; I'm hoping this year to incorporate 
exhibit's key features so that people can migrate painlessly from 
exhibit to mavo.  And mavo *does* support sheets v4.

**Comment by jimi75** [comment]
@karger Thanks for the answer! I will take a look at mavo.

**Comment by karger** [comment]
Mavo lets you write powerful interactive web applications that both read 
*and write* data.  You can find demos at https://mavo.io/demos/ .  While 
it provides some filtering and sorting capabilities, at doesn't yet have 
the rich visualizations of exhibit (maps and timelines).  We're looking 
at adding those this year.   If all you want to present is a list of 
items, then Mavo might already be able to work for you.  You can see 
interactive filtering on mavo's own documentation page, 
https://mavo.io/docs/functions/ 
<https://mavo.io/docs/functions/?fbclid=IwAR0tTn86McdoEz7JyRmfdhPk63FKSEKYfVpLbGYfnLb1XG8KPWGtva_hCJc> 
.  A more sophisticated one is here 
https://drafts.csswg.org/issues?spec=css-images-3&doc=cr-2012 
<https://drafts.csswg.org/issues?spec=css-images-3&doc=cr-2012&fbclid=IwAR0OzWr1e8kHYrIdmrlJDsUFPYOWfIRDPvIrJkhvfIoNqZuhQZffi1Kkvyc>

**Comment by lavinesp** [comment]
Thank you for responding.  My project displays a history of faculty of our school of music from about 1860 through the current time.  Each entry in a Google sheet includes names, years of service, discipline, rank, degrees, and a URL reference to a photo.  The data in the project is filterable and sortable and has info for about 600 faculty.  While the timeline feature in Exhibit was nice, it did not function correctly until you invoked it on a subset filter.  Do you think Mavo might work for my purposes?

**Comment by karger** [comment]
we don't have a timeline implemented in mavo yet.  i have recruited a 
student who I hope will be able to implement it this year.   an easy 
stopgap would be to export the spreadsheet to a csv; exhibit can read 
those just fine.  You'd have to re-export each time you add a faculty, 
but I suspect that doesn't happen all too often.

**Comment by lmorillas** [comment]
Mavo has no advanced visualization widget for now, but I think you can use
other d3js visualizations on it.
Is your data online to take a look and suggest a visualization?

**Comment by lavinesp** [comment]
Yes, the cvs option might work although we do have a fair amount of turnover each year.  The Google sheet with the data can be viewed at: https://docs.google.com/spreadsheets/d/e/2PACX-1vTwqLTg775_SRVewO3wVolAu966FzY6a2qIbulnHvHd-SzkWZFMReg-QI7YLzJ0w_CzE4l-wyzhn5JT/pubhtml

The broken faculty project URL is http://www2.potsdam.edu/cranehistory/faculty.html

It is part of a greater project exploring the history of our school of music located at  http://www2.potsdam.edu/cranehistory/

The history timeline chapters make use of TimelineJS developed at Northwestern University.  I was able to get those chapters functioning again after the deprecation of Sheets API v3.

**Comment by lmorillas** [comment]
And why not use the csv output from the google spreadsheet? I don't
understand the

<link rel="exhibit-data"
type="application/jsonp"
href="https://spreadsheets.google.com/feeds/list/0AoHrg7jeuwcYdE03NFNlSG51NlNvN1BWZlNwZGc0TEE/od6/public/basic?alt=json-in-script"
ex:converter="googleSpreadsheets" />

Use the csv output better.

And you are using exhibit2 syntax with exhibit3 release. And you are
not using the latest stable release of Exhibit 3. I could rewrite some
code suggestions in 2 or 3 days.

Saludos,

-- luismiguel  ***@***.***)

Saludos,

-- luismiguel  ***@***.***)

**Comment by jaygray0919** [comment]
> Mavo lets you write powerful interactive web applications that both read *and write* data.  You can find demos at https://mavo.io/demos/ .  While it provides some filtering and sorting capabilities, at doesn't yet have the rich visualizations of exhibit (maps and timelines).  We're looking at adding those this year.   If all you want to present is a list of items, then Mavo might already be able to work for you.  You can see interactive filtering on mavo's own documentation page, https://mavo.io/docs/functions/ <https://mavo.io/docs/functions>

@karger David, what can we do to help transition Exhibit-3 functions to Mavo?
For example, we have a number of Exhibit-3 instances that could serve as use cases for testing migration completeness.
We also could work on recoding Ex-3 components to work with a Mavo platform.
A valuable Ex-3 feature (important to us) is a facet-controlled timeline (Simile timeline + facet control/filters as implemented in "Composers")
Another is support for `flotr` extensions.
Another is support for very large datasets (10+ facets; 7,000+ terms per facet).
Ready-willing-able to help add value to Mavo and to help build seamless Ex-3-to-Mavo migration processes.
